chord/chordServerImpl.go

```diff
@@ -11,7 +11,7 @@ import (
 func (chordNode *ChordNode) GetPredecessor(ctx context.Context, e *empty.Empty) (*Node, error) {
 	pred, doesExist := chordNode.getPredecessor()
 	if !doesExist {
-		return nil, errors.New("predecessor is nil")
+		return nil, &nilPredecessor{}
```

Comment:
According to the function return signature (`(*Node, error)`) , you should `return &nilPredecessor{}, nil`

Reply:
So, `nilPredecessor` struct is a custom error (so that I don't need to initialize a new error by `errors.New('predecessor' is nil)`)
It's declared as such:
```
type nilPredecessor struct{} 
// the struct can be empty but it should implement error interface as below
func (n *nilPredecessor) Error() string {
   return "predecessor is nil"
}
```